File: src/Chapter-3-C99/Questions/Question2.c
//Write a program that formats product information entered by the
//user. A session with the program should look like this:
//
//Enter item number: 583
//Enter unit price: 13.5
//Enter purchase data (mm/dd/yyyy):  10/24/2010
//
//Item          Unit            Purchase
//              Price           Data
//583           $ 13.50         10/24/2010
#include <stdio.h>

int main(void)
{
    int item_num, day, month, year;
    float price;
    printf("Enter item number: ");
    scanf("%d", &item_num);
    printf("Enter unit price: ");
    scanf("%f", &price);
    printf("Enter purchase data (dd/mm/yyyy): ");
    scanf("%d/%d/%d", &day, &month, &year);
    printf("Item\t\tUnit\t\tPurchase\n");
    printf("\t\tPrice\t\tData\n");
    printf("%d\t\t$%.2f\t\t%d/%d/%d", item_num, price, day, month, year);

    return 0;
}

File: src/Chapter-3-Rust/Questions/Question2.rs
//Write a program that formats product information entered by the
//user. A session with the program should look like this:
//
//Enter item number: 583
//Enter unit price: 13.5
//Enter purchase date (mm/dd/yyyy): 10/24/2010
//
//Item		Unit		Purchase
//			Price		Date
//583		$13.50		10/24/2010
#[macro_use] extern crate text_io;
use std::process::Command;
use std::io::{self, Write};

fn main()
{
	let (item_num, day, month, year) : (i32, i32, i32, i32);
	let price : f32;
	print!("Enter item number: ");
	//show to screen rather than print and get lost in the magical buffer
	io::stdout().flush().unwrap(); //flush away xmas eve hangover and unwrap your presents
	scan!("{}", item_num);
	
	print!("Enter unit price: ");
	//Buffer!! - there must be a better way
	io::stdout().flush().unwrap();
	scan!("\n{}", price); //dont forget the \n
	
	print!("Enter purchase date (dd/mm/yyyy): ");
	//My goddess there is a lot of buffer
	io::stdout().flush().unwrap();
	scan!("\n{}/{}/{}", day, month, year); //at least no ampersands
	
	print!("Item\t\tUnit\t\tPurchase\n");
	print!("\t\tPrice\t\tDate\n"); 
	print!("{0}\t\t${1:.2}\t\t{2}/{3}/{4}\n", item_num, price, day, month, year);
	io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
